src/app/core/services/configuration/configuration.service.ts

```diff
@@ -24,6 +24,7 @@ export class ConfigurationService {
     return this.apiService
       .get(`configurations`, {
         headers: this.configHeaders,
+        runExclusively: true,
```

Comment:
Otherwise products data is being fetched in parallel without the selected currency on some pages. If those calls wait for the correct currency being set by the configurations call, everything works nicely